buildSrc/src/main/java/GitStatsTask.kt

```diff
@@ -0,0 +1,77 @@
+import com.esotericsoftware.yamlbeans.YamlConfig
+import com.esotericsoftware.yamlbeans.YamlWriter
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import java.io.FileInputStream
+import java.io.FileWriter
+import java.io.StringWriter
+import java.nio.charset.StandardCharsets
+
+/** Counts the occurance of changes in git commits for a given regex and
+ *  writes the result in a YML file. */
+open class GitStatsTask : DefaultTask() {
+    @get:Input lateinit var targetFile: String
+    @get:Input lateinit var commitFileFilter: Regex
+    @get:Input var binaryScore: Int = 0
+    @get:Input var commitSkipList: Array<String> = arrayOf()
+  
+    @TaskAction fun run() {
+        val countsByName = mutableMapOf<String, Int>()
+        val countsByCommit = mutableMapOf<String, Int>()
+        Runtime.getRuntime().exec("git log --no-merges --pretty='%an'%n%H --numstat").inputStream.bufferedReader().useLines { lines ->
+            var name = ""
+            var commit = ""
+            var commitNext = false
+            var skipNext = false
+            for(line in lines) {
+                if (line.startsWith('\'')) {
```

Comment:
I don't get it, what lines start with `'`? (Maybe add a comment)

Reply:
https://github.com/streetcomplete/StreetComplete/blob/1844ef7a17e3ed7c411316f918f1d2d0d304544b/buildSrc/src/main/java/GitStatsTask.kt#L22
The author line does. This was your code, I've fudged it a bit since. I suspect there is probably a better way to handle it like maybe separating with nulls or some sort of FSM per line or something.